bentoml/deployment/sagemaker/__init__.py

```diff
@@ -716,7 +716,15 @@ def delete(self, deployment_pb):
                     "AWS delete endpoint response: %s", delete_endpoint_response
                 )
             except ClientError as e:
-                raise _aws_client_error_to_bentoml_exception(e)
+                # If Sagemaker endpoint doesnt exist, we won't need to raise exception.
+                error_response = e.response.get('Error', {})
+                error_code = error_response.get('Code', 'Unknown')
+                error_message = error_response.get('Message', 'Unknown')
+                if (
+                    not error_code == 'ValidationException'
+                    or 'Could not find endpoint' not in error_message
+                ):
+                    raise _aws_client_error_to_bentoml_exception(e)
 
             _try_clean_up_sagemaker_deployment_resource(deployment_pb)
```

Comment:
as discussed offline, let's add function `delete_sagemaker_endpoint_if_exist`, and call this function in  `_try_clean_up_sagemaker_deployment_resource`.  

rename _try_clean_up_sagemaker_deployment_resource to _delete_sagemaker_deployment_resource_if_exist